src/main.rs

```diff
@@ -175,6 +175,9 @@ impl Results {
     }
 }
 
+// This is not yet designed to handle combinations of different parameters.
+// Combining more than one set of parameters, e.g. versions and ciphers, might
+// result in combinatorial explosion, as intended. But that's not guaranteed.
```

Comment:
So this is not deterministic? :fearful:

Reply:
I wouldn't say that. It's just that I find it hard to predict how exactly it will behave if a test case includes both a range of Versions and a list of ciphers as shared parameters. It will behave in a consistent way, it just might not be the intended one. It certainly needs some more work to become more consistent and also more flexible to handle paramteres other than version and cipher, without having to adjust the code everytime a new type of paramters is used.

Reply:
https://github.com/jallmann/tls-interop/issues/11